src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1727 @@
+        case GT_LT:
+        case GT_LE:
+        case GT_GE:
+        case GT_GT:
+        case GT_JCMP:
+            if (!varTypeIsFloating(tree->gtGetOp1()))
+            {
+                // We need two registers: tmpRegOp1 and tmpRegOp2
+                buildInternalIntRegisterDefForNode(tree);
+                buildInternalIntRegisterDefForNode(tree);
+                buildInternalRegisterUses();
+            }
+            srcCount = BuildCmp(tree);
+            break;
+
+        case GT_CKFINITE:
+            srcCount = 1;
+            assert(dstCount == 1);
+            buildInternalIntRegisterDefForNode(tree);
+            BuildUse(tree->gtGetOp1());
+            BuildDef(tree);
+            buildInternalRegisterUses();
+            break;
+
+        case GT_CMPXCHG:
+        {
+            NYI_LOONGARCH64("-----unimplemented on LOONGARCH64 yet----");
+
+            GenTreeCmpXchg* cmpXchgNode = tree->AsCmpXchg();
+            srcCount                    = cmpXchgNode->gtOpComparand->isContained() ? 2 : 3;
+            assert(dstCount == 1);
+
+            // if (!compiler->compOpportunisticallyDependsOn(InstructionSet_Atomics))
+            {
+                // For LOONGARCH exclusives requires a single internal register
+                buildInternalIntRegisterDefForNode(tree);
+            }
+
+            // For LOONGARCH exclusives the lifetime of the addr and data must be extended because
+            // it may be used multiple during retries
+
+            // For LOONGARCH atomic cas the lifetime of the addr and data must be extended to prevent
+            // them being reused as the target register which must be destroyed early
+
+            RefPosition* locationUse = BuildUse(tree->AsCmpXchg()->gtOpLocation);
+            setDelayFree(locationUse);
+            RefPosition* valueUse = BuildUse(tree->AsCmpXchg()->gtOpValue);
+            setDelayFree(valueUse);
+            if (!cmpXchgNode->gtOpComparand->isContained())
+            {
+                RefPosition* comparandUse = BuildUse(tree->AsCmpXchg()->gtOpComparand);
+
+                // For LOONGARCH exclusives the lifetime of the comparand must be extended because
+                // it may be used used multiple during retries
+                // if (!compiler->compOpportunisticallyDependsOn(InstructionSet_Atomics))
+                {
+                    setDelayFree(comparandUse);
+                }
+            }
+
+            // Internals may not collide with target
+            setInternalRegsDelayFree = true;
+            buildInternalRegisterUses();
+            BuildDef(tree);
+        }
+        break;
+
+        case GT_LOCKADD:
+        case GT_XORR:
+        case GT_XAND:
+        case GT_XADD:
+        case GT_XCHG:
+        {
+            NYI_LOONGARCH64("-----unimplemented on LOONGARCH64 yet----");
+
+            assert(dstCount == (tree->TypeGet() == TYP_VOID) ? 0 : 1);
+            srcCount = tree->gtGetOp2()->isContained() ? 1 : 2;
+
+#if 0
```

Comment:
Because the `InstructionSet_Atomics` is generated by scripts and it is related some Csharp files which we will submit them by a seperately PR.
For this PR, the related atomic instruction are not submitted.

Reply:
If that's the case, can you please introduce the new code in the separate PR?